include/swift/Remote/MetadataReader.h

```diff
@@ -2623,11 +2623,15 @@ class MetadataReader {
     // WARNING: the following algorithm works on current modern Apple
     // runtimes but is not actually ABI.  But it is pretty reliable.
 
+#if SWIFT_OBJC_INTEROP
     StoredPointer dataPtr;
     if (!Reader->readInteger(RemoteAddress(classAddress +
                                TargetClassMetadata<Runtime>::offsetToData()),
                              &dataPtr))
       return StoredPointer();
+#else
+    StoredPointer dataPtr = SWIFT_CLASS_IS_SWIFT_MASK;
```

Comment:
You can just `return StoredPointer();` here. You'll end up hitting the `if (!dataPtr)` path a few lines later anyway.